yarn-project/p2p/src/attestation_pool/memory_attestation_pool.ts

```diff
@@ -41,14 +45,27 @@ export class InMemoryAttestationPool implements AttestationPool {
     return Promise.resolve();
   }
 
+  public deleteAttestationsForSlotAndProposal(slot: bigint, proposalId: string): Promise<void> {
+    const slotAttestationMap = this.attestations.get(slot);
+    if (slotAttestationMap) {
+      slotAttestationMap.delete(proposalId);
+      this.log.verbose(`Removed attestation for slot ${slot}`);
+    }
+    return Promise.resolve();
+  }
+
   public deleteAttestations(attestations: BlockAttestation[]): Promise<void> {
```

Comment:
Similar to above, not clear to me how often we would use lower granularity than deleting for a slot at a time 🤷.